CMakeLists.txt

```diff
@@ -1,28 +1,22 @@
 cmake_minimum_required(VERSION 3.15)
 
-project(learnOpenGL)
+project(
+    learnOpenGL
+    VERSION 0.1.0
+    DESCRIPTION "Small OpenGL projects for study purpose"
+    HOMEPAGE_URL "https://github.com/unsip/learnOpenGL"
+    LANGUAGES CXX
+  )
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
-
-set(LIB_DIR "${CMAKE_CURRENT_SOURCE_DIR}/lib")
-set(SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/src")
-
-# Always prefer vendor-neutral GL lib.
-set(OpenGL_GL_PREFERENCE "GLVND")
 
 # GLFW3
-find_package(PkgConfig REQUIRED)
-pkg_search_module(GLFW REQUIRED glfw3)
-include_directories(${GLFW_INCLUDE_DIRS})
```

Comment:
Also, try to avoid variables whenever its possible. For this particular library use its provided CMake configuration files.